### PR TITLE
Classify EVM InvalidFEOpcode halts as contract reverts

### PR DESCRIPTION
**Pull request by jmg-duarte**
# Description

`is_contract_revert` failed to recognize EVM halts (like the `INVALID`/0xFE opcode older Solidity contracts emit on a missing selector) as contract-level rejections. anvil/revm surface these as `EVM error <HaltReason>` JSON-RPC errors, with no revert data and no "revert" in the message — so calls that deterministically fail at the bytecode level (e.g. `asset()` on Bancor BNT) were misclassified as transient transport failures.

In the EIP-4626 native price estimator this meant such tokens were never pinned as non-vault, so every native-price request triggered a fresh RPC round-trip that was guaranteed to fail.

Extra context: I got a list of the top 150 tokens in Ethereum through Dune, ran the asset call against them and basically summed up the responses here. Older contracts fail with this InvalidFEOpcode, which is what currently blocks GNO from being quoted in staging (i.e. when eip4626 is enabled)

# Changes

* Extend `is_contract_revert` to match the `InvalidFEOpcode` error — every halt is a deterministic contract-level rejection, not a transport issue to retry.
* Add a unit test (`contract_revert_accepts_evm_halt_errors`) covering `InvalidFEOpcode`, `OpcodeNotFound`, and `InvalidJump`.
* Add a forked-mainnet test that calls `asset()` across a real EIP-4626 vault (sUSDe) and several non-vault shapes — USDC (empty revert), WETH / stETH (plain reverts), and GNO / BNT (INVALID opcode) — asserting each non-vault is correctly classified as a contract revert.
* Add `BNT`, `stETH`, and `sUSDe` to `testlib::tokens` alongside the existing mainnet addresses; the new test consumes them from there.

## How to test

1. Unit tests: `cargo nextest run -p ethrpc contract_revert`
2. Forked test (requires mainnet fork URL):
   ```
   FORK_URL_MAINNET= cargo nextest run -p e2e
     forked_node_mainnet_asset_call_is_contract_revert
     --test-threads 1 --run-ignored ignored-only --failure-output final
3. Staging

Note, this is reproducible in staging, try trading a GNO token for example and you'll see liquidity errors.

### CODE REVIEW

**Review by m-sz — APPROVED** [review]

File: crates/ethrpc/src/alloy/errors.rs [hunk 1]
```diff
@@ -59,15 +59,26 @@ impl ContractErrorExt for ContractError {
...
             ContractError::TransportError(RpcError::ErrorResp(err)) => {
+                let message = err.message.to_lowercase();
                 err.as_revert_data().is_some()
                     // https://github.com/ethereum/go-ethereum/blob/8e2107dc39dc9dab132150ec915e7ac299f9eb48/internal/ethapi/errors.go#L42-L46
                     // https://github.com/alloy-rs/alloy/blob/b6753088241a50730c092bdba7036f52887c4c57/crates/rpc-types-eth/src/error.rs#L32
                     || err.code == 3
-                    || err.message.to_lowercase().contains("revert")
+                    || message.contains("revert")
+                    // anvil/revm surfaces halt reasons as
+                    // `EVM error <HaltReason>`. Match only the `INVALID`
```
Comment (MartinquaXD) [inline crates/ethrpc/src/alloy/errors.rs:77]: Did you check how other clients report those errors (e.g. geth, op-geth, nethermind)?

**Review by MartinquaXD — APPROVED** [review]
Good tests.
The improvement is useful enough as is to merge it but would be great to double check that this error handling also works on other clients.

File: crates/ethrpc/src/alloy/errors.rs (hunk 1 above)
Comment (jmg-duarte) [reply crates/ethrpc/src/alloy/errors.rs:77]: I didn't, I'll try it out before merging

Comment (jmg-duarte) [reply crates/ethrpc/src/alloy/errors.rs:77]: Tested with Tenderly and Geth, both worked
